lib/wpc/msap.c

```diff
@@ -448,6 +453,85 @@ int msap_scratchpad_block_read_request(uint32_t start_address, uint8_t number_of
     return confirm.payload.msap_image_block_read_confirm_payload.result;
 }
 
+int msap_config_data_item_set_request(const uint16_t endpoint,
+                                      const uint8_t *const payload,
+                                      const uint8_t payload_size)
+{
+    if (payload_size > MAXIMUM_CDC_ITEM_PAYLOAD_SIZE)
+    {
+        LOGE("Too large payload size (%d) for config data item\n", payload_size);
+        return WPC_INT_WRONG_PARAM_ERROR;
+    }
+
+    wpc_frame_t request = {
+        .primitive_id = MSAP_CONFIG_DATA_ITEM_SET_REQUEST,
+        .payload_length = sizeof(msap_config_data_item_set_req_pl_t),
```

Comment:
So even if CDD is 10 bytes, 80 bytes are sent? (padded with 00)
I guess you did it this way as it was the expectation on dualmcu app side.

I guess it is like this to mimic the old implementation of app_config :-)